src/browser-polyfill.js

```diff
@@ -6,7 +6,16 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 "use strict";
 
-if (typeof browser === "undefined") {
+var _supportsPromises = false;
+try {
+  _supportsPromises = browser.runtime.getPlatformInfo() instanceof Promise;
+} catch (e) {
+  console.log("promises not supported.");
+}
+
+if (typeof browser === "undefined" || !_supportsPromises) {
```

Comment:
A better option would be to change this line to:
```javascript
if (typeof browser === "undefined" || !(() => {
  let supportsPromises = false;
  try {
    supportsPromises = browser.runtime.getPlatformInfo() instanceof Promise;
  } catch (e) {
    console.log("promises not supported.");
  }
  return supportsPromises;
})()) {
```
And remove the previous block.